Dashboard/ViewModels/MainWindowViewModel.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Avalonia.Controls;
+using CommunityToolkit.Mvvm.ComponentModel;
+using Dashboard.Models;
+using Dashboard.Utils;
+using Dashboard.Views;
+using Microsoft.Extensions.DependencyInjection;
+
+
+
+namespace Dashboard.ViewModels
+{
+    public partial class MainWindowViewModel : ViewModelBase, IDisposable
+    {
+        /// <summary>
+        /// Observable variables that utilize ObservableObject for reactive commands.
+        /// </summary>
+        [ObservableProperty]
+        private object? _currentPage;
+        [ObservableProperty]
+        private ListItemTemplate _selectedListItem;
+        /// <summary>
+        /// Dictionary of the Views with their ViewModel and shorthand names.
+        /// </summary>
+        private readonly Dictionary<string, (UserControl View, ViewModelBase? ViewModel)> _views;
+
+        /// <summary>
+        /// Service Provider so the ViewModels can be created with their appropriate services required.
+        /// </summary>
+        private readonly IServiceProvider _serviceProvider;
+
+        /// <summary>
+        ///  Main constructor, creates a Dict with the ViewModels and their names,
+        /// _serviceProvider to allow dependency injection for ViewModels as needed,
+        /// and sets the first page
+        /// </summary>
+        public MainWindowViewModel()
+        {
+            _views = new Dictionary<string, (UserControl, ViewModelBase?)>();
+            _serviceProvider = DependencyInjection.ConfigureServices();
```

Comment:
q: does this mean we have 2 sets of services? how does DI handle this being called twice within an app?

Reply:
Ah I thought I removed the services in App.axaml.cs.It should have only be in MainWindowViewModel so it can handle the creation of the ViewModels but if we want to keep it in App.axaml.cs we can make it pass the services and use that instead of creating one in MainWindowViewModel.

Reply:
19b1f940a5ff9e1743d7213780f9944f81b4fd27

Reply:
idk which way is the best way to do it, how does testing work if we configureservices within the main window? I don't think it will change much, considering we can test each view individually without using the main window?

Reply:
We are able to test the views individually, you would just need to set the data context. 
See TestDataStore().